### ISSUE TITLE
infix_spaces_linter lacks a way to specify different rules for different `=` (`EQ_ASSIGN`, `EQ_SUB`, and `EQ_FORMALS`)

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1977]
Some authors (e.g. data.table) apply different rules for different `=`, for example:

```r
f = function(a=1) a
f(a=2)
```

Specifically, `EQ_SUB` and `EQ_FORMALS` should not have surrounding spaces, but `EQ_ASSIGN` should.

Currently, `exclude_operators` only supports string inputs, so this type of node-level differentiation is impossible.

There are two options:

(1) Allow specifying node names in `exclude_operators=`. `exclude_operators = c("EQ_SUB", "EQ_FORMALS")` is unambiguously a `parse_tag`. The advantage is a simplified user interface, the disadvantage is overloading.
(2) Add a new argument `exclude_parse_tags=`, which will match against `lintr:::infix_metadata$parse_tag` instead of `lintr:::infix_metadata$string_value`.

**Comment by AshesITR** [comment]
I'm slightly leaning towards (1) so as to not add two arguments that can do "the same". Are we exposing the xml tag names somewhere, so users can learn about them without studying xmlparsedata?

**Comment by MichaelChirico** [comment]
That feels like exposing some implementation details... e.g. we don't expose `lintr:::style_regexes` for `object_name_linter()`.

I'd suggest pointing to `utils::getParseData()` and accepting the parser node names instead. Otherwise we could point to `xmlparsedata:::xml_parse_token_map` as a more canonical source.

**Comment by AshesITR** [comment]
Okay; "EQ_SUB", "EQ_FORMALS" and "EQ_ASSIGN" are the names for both, `getParseData()` and `xmlparsedata()`.
Pointing to `getParseData()` will only differ in cases without ambiguity.
